op-supervisor/supervisor/backend/syncnode/node.go

```diff
@@ -333,6 +335,49 @@ func (m *ManagedNode) resetSignal(errSignal error, l1Ref eth.BlockRef) {
 	}
 }
 
+func (m *ManagedNode) resolveConflict(ctx context.Context, l1Ref eth.BlockRef, u eth.BlockID, f eth.BlockID) error {
+	// First try to reset to the last known safe block
+	s, err := m.backend.SafeDerivedAt(ctx, m.chainID, l1Ref.ID())
+	if err != nil {
+		return fmt.Errorf("failed to retrieve safe block for %v: %w", l1Ref.ID(), err)
+	}
+	if err := m.attemptReset(ctx, u, s, f); err == nil {
+		return nil
+	} else if !errCanBeWalkedBack(err) {
+		return fmt.Errorf("error during reset: %w", err)
+	}
+
+	// If initial attempt fails, try walking back
+	return m.walkBackToCommonAncestor(ctx, s.Number, u, f)
+}
+
+func (m *ManagedNode) attemptReset(ctx context.Context, unsafe eth.BlockID, safe eth.BlockID, finalized eth.BlockID) error {
+	m.log.Debug("Attempting reset", "unsafe", unsafe, "safe", safe, "finalized", finalized)
+	return m.Node.Reset(ctx, unsafe, safe, finalized)
+}
+
+func (m *ManagedNode) walkBackToCommonAncestor(ctx context.Context, start uint64, unsafe, finalized eth.BlockID) error {
```

Comment:
You could add "_linear" to this function name to indicate it could be replaced with more efficient searchers